src/engine/QueryPlanner.cpp

```diff
@@ -2893,21 +2907,25 @@ void QueryPlanner::GraphPatternPlanner::graphPatternOperationVisitor(Arg& arg) {
     // the clause.
     std::optional<ParsedQuery::DatasetClauses> datasetBackup;
     std::optional<Variable> graphVariableBackup = planner_.activeGraphVariable_;
+    auto defaultGraphBehaviourBackup = planner_.defaultGraphBehaviour_;
```

Comment:
1. Is this backup ever restored?
2.  Can you fix all those backups by making the resetting of them an automatic cleanup (with some generic backup factory), s.t. this code gets cleaner?

Reply:
I see the restoring, but I still like my suggestion about this being automatic.